wheatley/row_generation/complib_composition_generator.py

```diff
@@ -101,7 +113,7 @@ def from_url(cls, url: str) -> RowGenerator:
         main_url_parts = main_url.split("/")
 
         try:
```

Comment:
Do we need any of this try block?
`main_url_parts[0]` and `[1]` come straight from `self.complib_url` (which is a class field)
 `comp_id` is just reading back and overwriting the int parameter `comp_id`

Reply:
I think it is needed, because we don't know where the URLs are coming from.  This function will be used when the user inputs a whole URL into either CLI Wheatley (currently not allowed but on the roadmap) or to the RR integrated Wheatley (again currently not allowed but on the roadmap), and it feels better to be able to provide a clear error message describing what was wrong with the URL.

Reply:
Ah, sorry I'd missed GitHub's collapsing of the lines above and thought we were in `__init__` not `from_url(cls, url: str)`. It makes more sense now.

Do the doc tests cover call into this with a url?

Reply:
Ah that makes sense!  We should probably add some unit tests for `from_url`, even though it's not currently possible to use it.  I'll do that now.